java/client/src/main/java/glide/managers/CommandManager.java

```diff
@@ -27,82 +19,55 @@ public class CommandManager {
     private final ChannelHandler channel;
 
     /**
-     * Async (non-blocking) get.<br>
-     * See <a href="https://redis.io/commands/get/">REDIS docs for GET</a>.
+     * Build a command and send.
      *
-     * @param key The key name
+     * @param command
+     * @param responseHandler - to handle the response object
+     * @return A result promise of type T
      */
-    public CompletableFuture<String> get(String key) {
-        return submitNewRequest(RequestType.GetString, List.of(key));
+    public <T> CompletableFuture<T> submitNewCommand(
+            Command command, RedisExceptionCheckedFunction<Response, T> responseHandler) {
+        // write command request to channel
+        // when complete, convert the response to our expected type T using the given responseHandler
+        return channel
+                .write(prepareRedisRequest(command.getRequestType(), command.getArguments()), true)
+                .thenApplyAsync(response -> responseHandler.apply(response));
     }
 
     /**
-     * Async (non-blocking) set.<br>
-     * See <a href="https://redis.io/commands/set/">REDIS docs for SET</a>.
+     * Build a protobuf command/transaction request object.<br>
+     * Used by {@link CommandManager}.
      *
-     * @param key The key name
-     * @param value The value to set
+     * @param command - Redis command
+     * @param args - Redis command arguments as string array
+     * @return An uncompleted request. CallbackDispatcher is responsible to complete it by adding a
+     *     callback id.
      */
-    public CompletableFuture<String> set(String key, String value) {
-        return submitNewRequest(RequestType.SetString, List.of(key, value));
-    }
+    private RedisRequestOuterClass.RedisRequest.Builder prepareRedisRequest(
+            Command.RequestType command, String[] args) {
+        RedisRequestOuterClass.Command.ArgsArray.Builder commandArgs =
+                RedisRequestOuterClass.Command.ArgsArray.newBuilder();
+        for (var arg : args) {
+            commandArgs.addArgs(arg);
+        }
 
-    /**
-     * Build a command and submit it Netty to send.
-     *
-     * @param command Command type
-     * @param args Command arguments
-     * @return A result promise
-     */
-    private CompletableFuture<String> submitNewRequest(RequestType command, List<String> args) {
-        return channel
-                .write(RequestBuilder.prepareRedisRequest(command, args), true)
-                .thenApplyAsync(this::extractValueFromGlideRsResponse);
+        return RedisRequestOuterClass.RedisRequest.newBuilder()
+                .setSingleCommand(
+                        RedisRequestOuterClass.Command.newBuilder()
+                                .setRequestType(mapRequestTypes(command))
+                                .setArgsArray(commandArgs.build())
+                                .build())
+                .setRoute(
```

Comment:
why are you setting a route? why is the default all nodes?

Reply:
Yes.  This is what worked for the prototype and works fine for the standalone.  
This will get updated when the cluster client is added in the following PR (where we start to define routes). 
I can attach a TODO here to make sure it gets updated in the follow-up.

Reply:
can't you just not set the route? only set the route when a route is passed by the user?